internal/services/v1/relationships.go

```diff
@@ -169,6 +182,16 @@ func (ps *permissionServer) WriteRelationships(ctx context.Context, req *v1.Writ
 			return nil, rewritePermissionsError(ctx, err)
 		}
 
+		// Ensure wildcard writes have no subject relation.
+		if update.Relationship.Subject.Object.ObjectId == tuple.PublicWildcard {
+			if update.Relationship.Subject.OptionalRelation != "" {
+				return nil, status.Errorf(
+					codes.InvalidArgument,
+					"wildcard (public) relationships require an empty subject relation",
+				)
+			}
+		}
```

Comment:
Is this something we should have in the `HandwrittenValidate()` in authzed-go?

Reply:
We could, although it does raise a question of when we should be items in there vs here.

Reply:
This is purely an invariant that is self contained within the contents of the request, thus I think it should be in a validation step, rather than in a handler where our logic lives.

Reply:
Okay. This will require another change to authzed-go, though

Reply:
Yep, I think that's the right way to do it.

Reply:
Handled in https://github.com/authzed/authzed-go/pull/47